app/views/groups/assign_scans.html.erb

```diff
@@ -0,0 +1,287 @@
+<% content_for :head do %>
+    <%= stylesheet_link_tag('clickable') %>
+    <%= javascript_include_tag 'Results/file_selector',
+                               'panes',
+                               'pdfjs',
+                               'testresults_config',
+                               'Results/context_menu' %>
+
+    <%= stylesheet_link_tag('grader') %>
+    <%= stylesheet_link_tag('pdfjs/viewer') %>
+    <%= javascript_tag do -%>
+        PDFJS.workerSrc = "<%= @host %>/assets/pdfjs_lib/pdf.worker.js";
+    <% end -%>
+    <%= render partial: 'layouts/head' %>
+<% end %>
+
+<!-- Keeping this here until to keep the autocomplete looking functional until the codebase is updated -->
+<link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
+<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
+
+<script>
+    $(document).ready(function() {
+        <%
+          num_valid = @assignment.get_num_valid
+          num_assigned = @assignment.get_num_assigned
+        %>
+        update_bar(<%= num_valid %>, <%= num_assigned %>);
+
+        $( "#names" ).autocomplete({
+            source: "get_names",
+            minLength: 1,
+            select: function(event, ui) {
+                $( "#student_id" ).val(ui.item.id);
+            }
+        });
+        $( "#assign_student" ).submit(function (e) {
+            e.preventDefault();
+            $.ajax({
+                method: 'GET',
+                url: "assign_student_and_next",
+                dataType: "json",
+                data: $( "#assign_student" ).serialize(),
+                success: function( data ) {
+                    parseNextScan( data )
+                }
+            });
+        });
+        function getNextScan() {
+            $.ajax({
+                method: 'GET',
+                url: "next_grouping",
+                dataType: "json",
+                data: {
+                    a_id: <%= @assignment.id %>
+                },
+                success: function( data ) {
+                    parseNextScan( data )
+                }
+            });
+        }
+        function parseNextScan(data) {
+            $( "#group_members" ).empty();
+            for(i = 0; i < data.students.length; i++){
+                $( "#group_members" ).append("<p>" + data.students[i] + "</p>")
+            }
+            current_group = $( "#grouping_id" ).val();
+            $( "#assign_student" ).find("input").val("");
+            if( current_group != data.grouping_id){
```

Comment:
I assume you're using this to determine whether the end has been reached (no more groups to assign). If so, you should just have the JSON returned by the server to be empty, and check for that.